content/collections.md

```diff
@@ -0,0 +1,370 @@
+
+The simple way to work around the problem is to take the application down for the period in between deployment and completing the migration. This is far from ideal, especially considering some migrations can take hours to run!
+
+A better approach is a multi-stage deployment. The basic idea is that:
+
+1. First you deploy a version of your application that can handle both the old and the new schema.
+  - In our case, it'd be code that doesn't expect the `todoCount` to be there, but which correctly updates it when new todos are created.
+2. Second you run the migration.
+  - At this point you should be confident that all lists have a `todoCount`.
+3. Finally, you deploy the new code that relies on the new schema and no longer copes with the old schema.
+  - Now we are safe to rely on `list.todoCount` in our UI.
+
+Another thing to be aware of, especially with such multi-stage deploys, is that being prepared to rollback is important! For this reason, the migrations package allows you to specify a `down()` function and set `MIGRATION=x` to migration _back_ to version `x`. 
+
+If you find you need to roll your code version back, you'll need to be careful about the data, and step careful through your deployment steps in reverse.
+
+## Relations between collections
+
+As we discussed earlier, it's very common in Meteor applications to relate documents in different collections. Consequently, it's also very common to need to write queries fetching related documents once you have a document you are interested (for instance all the todos that are on a single list).
+
+To do so, we can attach functions to the prototype of the documents that belong to a given collection, to give us "methods" on the documents (in the object oriented sense). We can then use these methods to create new queries to find related documents.
+
+### Collection Helpers
+
+To do, we can use the [`dburles:collection-helpers`](https://atmospherejs.com/dburles/collection-helpers) to easily attach such methods (or "helpers") to documents. For instance:
+
+```js
+Lists.helpers({
+  // A list is considered to be private if it has a userId set
+  isPrivate() {
+    return !!this.userId;
+  }
+});
+```
+
+Once we've attached this helper to the `Lists` collection, every time we fetch a list from the database (on the client or server), it will have a `.isPrivate()` helper available:
+
+```js
+const list = Lists.findOne();
+if (list.isPrivate()) {
+  console.log('The first list is private!');
+}
+```
+
+### Relational helpers
+
+Now we can attach helpers to documents, it's simple to define a helper that fetches related documents
+
+// XXX: we don't actually have this in the Todos app, but we probably should
+// XXX: is this .todos() or .getTodos() ?
```

Comment:
So bikesheddy. I prefer `todos()` but you are the naming guy, so let's do `getTodos()`?